src/open_inwoner/openzaak/clients.py

```diff
@@ -692,43 +702,207 @@ def fetch_open_tasks(self, bsn: str) -> list[OpenTask]:
         return results
 
 
+TClient = TypeVar("TClient", bound=APIClient)
+
+
+@dataclass(frozen=True)
+class ZgwClientResponse:
+    """A single response in a MultiZgwClientResult."""
+
+    client: TClient
+    result: Any
+    exception: Exception | None = None
+
+
+@dataclass(frozen=True)
+class MultiZgwClientProxyResult:
+    """Container for a multi-backend responses"""
+
+    responses: list[ZgwClientResponse]
+
+    @cached_property
+    def has_errors(self) -> bool:
+        return any(r.exception is not None for r in self.responses)
+
+    @cached_property
+    def failing_responses(self) -> list[ZgwClientResponse]:
+        return list(r for r in self if r.exception is not None)
+
+    @cached_property
+    def successful_responses(self) -> list[ZgwClientResponse]:
+        return list(r for r in self if r.exception is None)
+
+    def raise_on_failures(self):
+        """Raise a MultiZgwClientProxyError wrapping all errors raised by the clients."""
+        if not self.has_errors:
+            return
+
+        raise MultiZgwClientProxyError([r.exception for r in self.failing_responses])
+
+    def join_results(self):
+        """Joins results for all successful responses in a list."""
```

Comment:
The previous docstring is imperative, this one declarative. We should stick to one standard. I prefer the imperative style, though don't have strong feelings about it.